test/e2e/join_and_leave_test.go

```diff
@@ -114,3 +121,51 @@ var _ = Describe("Test member cluster join and leave flow", Ordered, Serial, fun
 		ensureCRPAndRelatedResourcesDeletion(crpName, allMemberClusters)
 	})
 })
+
+var _ = Describe("Test member cluster force delete flow", Ordered, Serial, func() {
+	Context("Test cluster join and leave flow with member agent down and force delete member cluster", Ordered, Serial, func() {
+		It("Simulate the member agent going down in member cluster", func() {
+			Eventually(func() error {
+				return updateMemberAgentDeploymentReplicas(0)
+			}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to simulate member agent going down")
+		})
+
+		It("Delete member cluster CR associated to the member cluster with member agent down", func() {
+			var mc clusterv1beta1.MemberCluster
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: memberCluster3WestProdName}, &mc)).To(Succeed(), "Failed to get member cluster")
+			Expect(hubClient.Delete(ctx, &mc)).Should(Succeed())
+		})
+
+		It("Should garbage collect resources owned by member cluster and force delete member cluster CR after force delete wait time", func() {
+			memberClusterNamespace := fmt.Sprintf(utils.NamespaceNameFormat, memberCluster3WestProdName)
+			Eventually(func() bool {
+				var ns corev1.Namespace
+				return apierrors.IsNotFound(hubClient.Get(ctx, types.NamespacedName{Name: memberClusterNamespace}, &ns))
+			}, time.Minute*3, eventuallyInterval).Should(BeTrue(), "Failed to garbage collect resources owned by member cluster")
+
+			Eventually(func() bool {
+				return apierrors.IsNotFound(hubClient.Get(ctx, types.NamespacedName{Name: memberCluster3WestProdName}, &clusterv1beta1.MemberCluster{}))
+			}, eventuallyDuration, eventuallyInterval).Should(BeTrue(), "Failed to force delete member cluster")
+		})
+	})
+
+	AfterAll(func() {
+		By("Simulate the member agent coming back up")
+		Eventually(func() error {
+			return updateMemberAgentDeploymentReplicas(1)
+		}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to simulate member agent coming back up")
+
+		createMemberCluster(memberCluster3WestProd.ClusterName, memberCluster3WestProd.PresentingServiceAccountInHubClusterName, labelsByClusterName[memberCluster3WestProd.ClusterName], annotationsByClusterName[memberCluster3WestProd.ClusterName])
+		checkIfMemberClusterHasJoined(memberCluster3WestProd)
+	})
+})
+
+func updateMemberAgentDeploymentReplicas(replicas int32) error {
```

Comment:
this function name doesn't mention that it updated a certain cluster agent replica to 0. Better pass the cluster in.